pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -443,6 +448,21 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 	}
 }
 
+// flushUnschedulableQLeftover moves pod which stays in unschedulableQ longer than the durationStayUnschedulableQ
+// to activeQ.
+func (p *PriorityQueue) flushUnschedulableQLeftover() {
+	p.lock.Lock()
+	var podsToMove []*v1.Pod
+	for _, pod := range p.unschedulableQ.pods {
+		if p.clock.Now().Sub(podTimestamp(pod).Time) > unschedulableQTimeInterval {
```

Comment:
Do not call "clock.Now()" in the loop. Call it once outside of the loop, store the result, and use the stored result in the loop.